source/read-write-concern/tests/README.rst

```diff
@@ -45,22 +45,43 @@ the spec is final.
 Format
 ------
 
+Connection String
+~~~~~~~~~~~~~~~~~
+
+These tests are designed to excersize the connection string parsing related
+to read concern and write concern.
+
 Each YAML file contains an object with a single ``tests`` key. This key is an
 array of test case objects, each of which have the following keys:
 
 - ``description``: A string describing the test.
 - ``uri``: A string containing the URI to be parsed.
+- ``valid:``: a boolean indicating if parsing the uri should result in an error.
 - ``writeConcern:`` A document indicating the expected write concern.
-- ``isAcknowledged:`` A boolean indicating whether the write concern is 
-  acknowledged.
 - ``readConcern:`` A document indicating the expected read concern.
-- ``error:``: a boolean indicating if parsing the uri should result in an error, or the string
-  'optional' indicating that reporting of an error is optional
 
-If a test case includes a null value for one of these keys,
-no assertion is necessary. This both simplifies parsing of the
-test files (keys should always exist) and allows flexibility for drivers that
-might substitute default values *during* parsing.
+If a test case includes a null value for one of these keys, or if the key is missing,
+no assertion is necessary. This both simplifies parsing of the test files and allows flexibility
+for drivers that might substitute default values *during* parsing.
+
+Document
+~~~~~~~~
+
+These tests are designed to ensure compliance with the spec in relation to what should be 
+sent to the server.
+
+Each YAML file contains an object with a single ``tests`` key. This key is an
+array of test case objects, each of which have the following keys:
```

Comment:
Same as above.. not all ``tests have the following keys``, but they probably have most of them